### ISSUE TITLE
Can i pass Excel file content instead of path to source Excel file ?

### ISSUE THREAD

**Issue by suman-gulapalemmu** [stevetarver/excel-as-json#5]
I am using multiparty node module to process the uploaded files and have XLSX file content as bytes. 

Can i pass that bytes content to `src` to API `convertExcel(<src>, <dst>, isColOriented, callback);` instead of path to source Excel file ?

@stevetarver Waiting for your reply man.

**Comment by stevetarver** [comment]
Nope. This excerpt from the processFile function
```coffeescript
  else
    excel src, (err, data) ->
      if err
        callback "Error reading #{src}: #{err}"
      else
        result = convert data, isColOriented
```
hints that the `excel` package is used to convert the source file to a 2D data grid which is passed to the convert function. Check out the [convert spec](https://github.com/stevetarver/excel-as-json/blob/master/spec/convertSpec.coffee) for examples of data requirements. 

I built this package for offline processing in `grunt-excel-as-json` with zero concern for processing speed. The `excel` package is easy to use, but as I recall, not so performant. If you like my approach to organizing json data in a table, you could use one of the many Excel packages to turn your bytes into a 2D array and pass it to the convert function.

**Comment by suman-gulapalemmu** [comment]
Hey @stevetarver, thanks for the reply man, i have got over the challenge which i mentioned. `multiparty` module by default writes the file to OS Temporary location and that location is accessible from this property `file.path` . 

Now am using that variable and passing it on to this code snippet :

```
convertExcel(file.path, undefined, false, function (err, data) {
            if (err) console.log('Error occured while processing excel document.');
            console.log('processing successfull !');
            console.log('Data is : ' + JSON.stringify(data));
        });
```

**Output :**

```
processing successfull !
Data is : []
```

I have re-read the documentation and followed all mandatory steps but still unable to figure out why i am getting this empty array output despite of having content in my excel sheet.

I have attached excel sheet (XLSX format) and it has `sheet 0` , please guide me where i am going wrong ?
[MyExcel.xlsx](https://github.com/stevetarver/excel-as-json/files/540782/MyExcel.xlsx)

**Comment by stevetarver** [comment]
`convertExcel()` takes a 2D array of data and a boolean indicating if the data is column or data oriented. Since you are using a file, you should use the `processFile()` function. Your args look right for processFile.

**Comment by suman-gulapalemmu** [comment]
@stevetarver can you please provide an example either in documentation or here, i tried processFile() function and its not working.

```
var convertExcel = require('excel-as-json').processFile();

convertExcel(file.path, undefined, false, function (err, data) {
            if (err) console.log('Error occured while processing excel document.');
            console.log('processing successfull !');
            console.log('Data is : ' + JSON.stringify(data));
        });
```

Am ending up with an exception as :

`TypeError: convertExcel is not a function`

**Comment by stevetarver** [comment]
The `()` at the end of `require('excel-as-json').processFile();` means invoke the function. You want a function reference. From the README.md:

```js
convertExcel = require('excel-as-json').processFile;
```

**Comment by stevetarver** [comment]
I'm not sure that you ever resolved your issues, but I have released a new version 1.0.4 that resolves some user issues getting [] back from reading a valid file.
